x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/index.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  getDefaultEuiMarkdownParsingPlugins,
+  getDefaultEuiMarkdownProcessingPlugins,
+} from '@elastic/eui';
+
+import * as timelineMarkdownPlugin from './timeline';
+
+export const uiPlugins = [timelineMarkdownPlugin.plugin];
+export const parsingPlugins = getDefaultEuiMarkdownParsingPlugins();
+export const processingPlugins = getDefaultEuiMarkdownProcessingPlugins();
+
+parsingPlugins.push(timelineMarkdownPlugin.parser);
+processingPlugins[1][1].components.timeline = timelineMarkdownPlugin.renderer;
```

Comment:
The `processingPlugins` array is initialized above by invoking the imported `getDefaultEuiMarkdownProcessingPlugins` function:

```
export const processingPlugins = getDefaultEuiMarkdownProcessingPlugins();
``` 

The type signature of `getDefaultEuiMarkdownProcessingPlugins` doesn't provide assurances that it will always return the same number of plugins in the same order.

Consider replacing the hard-coded index `1`  (in `processingPlugins` on the line below):

```
processingPlugins[1][1].components.timeline = timelineMarkdownPlugin.renderer;
```

with a call to [array.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), where the lambda passed to `find` contains criteria, like a specific plugin ID, that will insulate this code from changes to the implementation of the `getDefaultEuiMarkdownProcessingPlugins()`.

Reply:
> `processingPlugins[1][1].components.timeline = timelineMarkdownPlugin.renderer;`

This line of code is TS-compatible and it will break if `[1][1]` change in the future.